src/core/server/http/integration_tests/__fixtures__/plugins/dummy_security/server/plugin.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import Boom from 'boom';
+import { Authenticate, CoreSetup } from '../../../../../../../../core/server';
+
+interface User {
+  id: string;
+  roles?: string[];
+}
+
+interface Storage {
+  value: User;
+  expires: number;
+}
+
+export const url = {
+  auth: '/auth',
+  authRedirect: '/auth/redirect',
+  exception: '/exception',
+  redirectTo: '/login',
+};
+
+export const sessionDurationMs = 30;
+export class DummySecurityPlugin {
+  public setup(core: CoreSetup) {
+    const authenticate: Authenticate<Storage> = async (request, sessionStorage, t) => {
+      if (request.path === url.authRedirect) {
+        return t.redirected(url.redirectTo);
+      }
+
+      if (request.path === url.exception) {
+        throw new Error('sensitive info');
+      }
+
+      if (request.headers.authorization) {
+        const user = { id: '42' };
+        sessionStorage.set({ value: user, expires: Date.now() + sessionDurationMs });
+        return t.authenticated({ credentials: user });
```

Comment:
@elastic/kibana-security are you going to use `req.auth` somewhere downstream? I think we can organise our own user credentials storage and don't relay on `hapi` mechanism. for now we can keep it

Reply:
We make use of `req.auth.credentials` within the security plugin, and it appears that the `ui_render_mixin` checks `request.auth.isAuthenticated` as part of its flow: https://github.com/legrego/kibana/blob/1e6270b61362ebc38215099f2bd811c7be2d936e/src/legacy/ui/ui_render/ui_render_mixin.js#L122

Reply:
Yes, that the way it's being used right now. As I understand we don't want to provide access to auth credentials to all plugins.
In New platform I can imagine that 3rd party plugin makes the same check as:
```js
const isUserAuthenticated = plugins.security ? plugins.security.isAuthenticated() : false;
```

Reply:
This `request.auth.isAuthenticated` usage in `ui_render_mixin.js` (OSS) looks very weird to me, and I'd ping the developer who added that to at least leave a comment what it's and why it's needed here so that we can better understand the use case  here :slightly_smiling_face: 

The problem is that OSS can't depend on x-pack plugins and hence security (even optionally, at least for the time being). And at the same time core now has some abstract knowledge about `authentication` (e.g. `http.registerOnAuth`) so theoretically it can tell whether request is authenticated or not, without giving any other details that would `security` provide. Maybe not exactly `isAuthenticated`, but more like `authenticatioState` that'd have 3 states: authenticated, not-authenticated, not-even-tried to be authenticated (when no `OnAuth` handler has been registered), but not quite sure. Maybe it'd make sense to gather all the use cases and discuss them one by one in a follow-up.

Reply:
I **suspect** it was done to help user to identify their logged/unlogged status visually.
agree lets solve this in follow ups